icon/icmodule/revision.go

```diff
@@ -115,73 +116,43 @@ const (
 	RevisionUpdatePRepStats = Revision22
 	RevisionBlockAccountAPI = Revision22
 
-	RevisionIISS4R0 = Revision24
+	RevisionIISS4R0            = Revision24
+	RevisionChainScoreEventLog = Revision24
 
-	RevisionIISS4R1      = Revision25
-	RevisionVoteEventLog = Revision25
+	RevisionIISS4R1 = Revision25
 )
 
-var revisionFlags = []module.Revision{
-	// Revision0
-	module.UseChainID | module.UseMPTOnEvents | module.UseCompactAPIInfo | module.LegacyFeeCharge | module.LegacyFallbackCheck | module.LegacyContentCount | module.LegacyBalanceCheck | module.LegacyNoTimeout,
-	// Revision1
-	0,
-	// Revision2
-	module.AutoAcceptGovernance,
-	// Revision3
-	module.LegacyInputJSON | module.LegacyFallbackCheck | module.LegacyContentCount | module.LegacyBalanceCheck,
-	// Revision4
-	0,
-	// Revision5
-	0,
-	// Revision6
-	0,
-	// Revision7
-	0,
-	// Revision8
-	0,
-	// Revision9
-	0,
-	// Revision10
-	0,
-	// Revision11
-	0,
-	// Revision12
-	0,
-	// Revision13
-	module.LegacyFeeCharge | module.LegacyNoTimeout,
-	// Revision14
-	module.LegacyInputJSON | module.InputCostingWithJSON,
-	// Revision15
-	0,
-	// Revision16
-	0,
-	// Revision17
-	0,
-	// Revision18
-	module.FixLostFeeByDeposit,
-	// Revision19
-	module.PurgeEnumCache,
-	// Revision20
-	module.FixMapValues,
-	// Revision21
-	module.MultipleFeePayers,
-	// Revision22
-	0,
-	// Revision23
-	module.FixJCLSteps,
-	// Revision24
-	0,
-	// Revision25
-	module.ReportDoubleSign,
+var revisionFlags []module.Revision
+
+var toggleFlagsOnRevision = []struct {
+	value int
+	flags module.Revision
+}{
+	{Revision0, module.UseChainID | module.UseMPTOnEvents | module.UseCompactAPIInfo | module.LegacyFeeCharge | module.LegacyFallbackCheck | module.LegacyContentCount | module.LegacyBalanceCheck | module.LegacyNoTimeout},
+	{Revision2, module.AutoAcceptGovernance},
+	{Revision3, module.LegacyInputJSON | module.LegacyFallbackCheck | module.LegacyContentCount | module.LegacyBalanceCheck},
+	{Revision13, module.LegacyFeeCharge | module.LegacyNoTimeout},
+	{Revision14, module.LegacyInputJSON | module.InputCostingWithJSON},
+	{Revision18, module.FixLostFeeByDeposit},
+	{Revision19, module.PurgeEnumCache},
+	{Revision20, module.FixMapValues},
+	{Revision21, module.MultipleFeePayers},
+	{Revision23, module.FixJCLSteps},
+	{Revision24, module.ReportConfigureEvents},
+	{Revision25, module.ReportDoubleSign},
```

Comment:
Why don't you use semantic revision name instead of numeric name like this:
Revision25 -> RevisionIISS4R1

Reply:
I prefer to use the revision name directly here, because it is more intuitive.